datascience/src/pipeline/parsers/flux/flux.py

```diff
@@ -0,0 +1,228 @@
+import logging
+import xml
+import xml.etree.ElementTree as ET
+import gzip
+import base64
+from datetime import datetime
+from functools import partial
+from typing import List
+from xml.etree.ElementTree import ParseError
+
+import pandas as pd
+
+from src.pipeline.parsers.flux.log_parsers import (
+    default_log_parser,
+    parse_coe,
+    parse_cox,
+    parse_dep,
+    parse_dis,
+    parse_far,
+    parse_lan,
+    parse_pno,
+    parse_rtp,
+)
+
+from src.utils.flux import NS_FLUX, get_element, get_type, get_purpose, get_text, make_datetime
+from src.pipeline.parsers.utils import tagged_children
+
+class FLUXParsingError(Exception):
+    """Raised when an FLUX message cannot be parsed."""
+
+    pass
+
+def parse_metadata(el: xml.etree.ElementTree.Element, op_type):
+    vessel = get_element(el,'ram:SpecifiedVesselTransportMeans')
+
+    metadata =  {"operation_type": op_type,
+            "operation_datetime_utc": make_datetime(get_text(el,'.//ram:CreationDateTime/udt:DateTime')),
+            "flux_id": get_text(el,'.//ram:ID[@schemeID="UUID"]'),
+            "flux_datetime_utc": make_datetime(get_text(el,'.//ram:AcceptanceDateTime/udt:DateTime')),
+            "cfr": get_text(vessel,'.//*[@schemeID="CFR"]'),
+            "ircs": get_text(vessel,'.//*[@schemeID="IRCS"]'),
+            "external_identification": get_text(vessel,'.//*[@schemeID="EXT_MARK"]'),
+            "vessel_name": get_text(vessel,'ram:Name'),
+            "flag_state": get_text(vessel,'.//ram:ID[@schemeID="TERRITORY"]'),
+            "imo": get_text(vessel,'.//*[@schemeID="UVI"]'),
+            "trip_number": get_text(el,'.//ram:ID[@schemeID="EU_TRIP_ID"]')
+            }
+    if op_type=='COR':
+        metadata= {**metadata, "referenced_flux_id": get_text(el,'.//ram:ReferencedID[@schemeID="UUID"]')}
+
+    return metadata
+
+def simple_parser(el: xml.etree.ElementTree.Element, op_type):
+    metadata =  parse_metadata(el,op_type)
+
+    children = tagged_children(el)
+    data_iter=[]
+    if "SpecifiedFishingActivity" in children:
+        for child in children["SpecifiedFishingActivity"]:
+            data = parse_(child)
+            data_iter.append(data)
+
+    return metadata, iter(data_iter)
+
+def parse_not(not_):
+    op_type = get_purpose(not_)
+    metadata = parse_metadata(not_, op_type)
+
+    children = tagged_children(not_)
+    data_iter=[]
+    if "SpecifiedFishingActivity" in children:
+        for child in children["SpecifiedFishingActivity"]:
+            msg_type = get_type(child)
+            if msg_type=='ARRIVAL' :
```

Comment:
Les messages de test de PNO (Prior notification of arrival to port (Reason=Shelter).xml et Prior notification of arrival to port (Reason=Landing).xml) ne sont pas correctement identifiés comme PNO.
Cela vient du fait que `get_type` ne renvoie pas 'ARRIVAL' mais 'RTP'.